### PR TITLE
REALMC-7341: Create integrated Realm client

### PR DESCRIPTION
**Pull request by makesitgo**

### CODE REVIEW

**Comment by makesitgo** [comment]
Ignore the `Add MustBeNil helper` commit changes, I'll be dropping those in favor of #28

File: internal/cloud/realm/auth.go [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+	var auth AuthResponse
+	var authErr error
+
+	dec := json.NewDecoder(res.Body)
+	defer func() {
+		if err := res.Body.Close(); err != nil {
+			authErr = err
+		}
+	}()
```
Comment (tsedgwick) [inline internal/cloud/realm/auth.go:46]: should this really be an authErr? I would just `defer res.Body.Close()`

File: internal/cloud/realm/auth.go [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+	if resErr != nil {
+		return AuthResponse{}, resErr
+	}
+
+	if res.StatusCode != http.StatusOK {
+		return AuthResponse{}, UnmarshalServerError(res)
+	}
+
+	var auth AuthResponse
+	var authErr error
```
Comment (tsedgwick) [inline internal/cloud/realm/auth.go:39]: if we remove that defer func() - we can just inline the err creation

**Review by tsedgwick — APPROVED** [review]
lgtm

File: internal/cloud/realm/auth.go (hunk 1 above)
Comment (makesitgo) [reply internal/cloud/realm/auth.go:46]: result of a bad linter setting, we're squared away here now

**Pull request merged**